src/index.ts

```diff
@@ -0,0 +1,275 @@
+	info(`Running Command: ${command}`);
+	execSync(command, { cwd: config["workingDirectory"], env: process.env });
+	info(`✅ Wrangler installed`);
+	endGroup();
+}
+
+function authenticationSetup() {
+	startGroup("🔐 Authenticating with Cloudflare");
+	try {
+		process.env.CLOUDFLARE_API_TOKEN = config["CLOUDFLARE_API_TOKEN"];
+		process.env.CLOUDFLARE_ACCOUNT_ID = config["CLOUDFLARE_ACCOUNT_ID"];
+		info(`Authentication process initiated with - API Token`);
+	} catch (error) {
+		setFailed(
+			`Authentication details were not found. Please input an 'apiToken' to the action.`,
+		);
+	}
+	endGroup();
+}
+
+async function execCommands(commands: string[], cmdType: string) {
+	if (!commands.length) {
+		return;
+	}
+
+	startGroup(`🚀 ${cmdType} Commands Group`);
+
+	const arrPromises = commands.map(async (command) => {
+		const npxCmd = command.startsWith("wrangler")
+			? `${getNpxCmd()} ${command}`
+			: command;
+
+		info(`🚀 Executing command: ${npxCmd}`);
+
+		await execAsync(npxCmd, {
+			cwd: config["workingDirectory"],
+			env: process.env,
+		});
+	});
+
+	await Promise.all(arrPromises);
+	endGroup();
+}
+
+/**
+ * A helper function to get the secret from the environment variables.
+ */
+const getSecret = (secret: string) => process.env[secret] ?? "";
+
+async function legacyUploadSecrets(
+	secrets: string[],
+	environment?: string,
+	workingDirectory?: string,
+) {
+	try {
+		const arrPromises = secrets
+			.map((secret) => {
+				const command = `${getSecret(
+					secret,
+				)} | ${getNpxCmd()} wrangler secret put ${secret}`;
+				return environment ? command.concat(`--env ${environment}`) : command;
+			})
+			.map(
+				async (command) =>
+					await execAsync(command, {
+						cwd: workingDirectory,
+						env: process.env,
+					}),
+			);
+
+		await Promise.all(arrPromises);
+	} catch {
+		setFailed(`🚨 Error uploading secrets`);
+	}
+}
+
+function semverCompare(version1: string, version2 = "3.4.0") {
+	const version1Parts = version1.split(".");
+	const version2Parts = version2.split(".");
+
+	for (const version1Part of version1Parts) {
+		const version2Part = version2Parts.shift();
+
+		if (version1Part !== version2Part && version2Part) {
+			return version1Part < version2Part ? true : false;
+		}
+	}
+
+	return false;
+}
+
+async function uploadSecrets() {
+	startGroup("🔑 Uploading Secrets");
+	const secrets: string[] = config["secrets"];
+	const environment = config["ENVIRONMENT"];
+	const workingDirectory = config["workingDirectory"];
+
+	if (!secrets.length) {
+		return;
+	}
+	if (semverCompare(config["WRANGLER_VERSION"]))
+		return legacyUploadSecrets(secrets, environment, workingDirectory);
+
+	const secretObj = secrets.reduce((acc: any, secret: string) => {
+		acc[secret] = getSecret(secret);
+		return acc;
+	}, {});
+
+	const environmentSuffix = !environment.length ? "" : ` --env ${environment}`;
+
+	const secretCmd = `echo ${JSON.stringify(
+		secretObj,
+	)} | ${getNpxCmd()} wrangler secret:bulk ${environmentSuffix}`;
```

Comment:
I didnt seem to have an issue with it locally testing or the E2E tests... Can you confirm this doesnt work?

Reply:
the error is masked because you're setting `stdio: "ignore"` and the child process doesn't cause an exception to be thrown because `wrangler secret:bulk` isn't returning a non-zero exit code on error.

<img width="656" alt="image" src="https://github.com/cloudflare/wrangler-action/assets/1395018/21b213d7-2870-464a-bbfe-7de29faf9d54">